src/pages/home/report/UserTypingEventListener.tsx

```diff
@@ -0,0 +1,71 @@
+import type {RouteProp} from '@react-navigation/native';
+import {useIsFocused, useRoute} from '@react-navigation/native';
+import {useEffect, useRef} from 'react';
+import {InteractionManager} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import Navigation from '@libs/Navigation/Navigation';
+import type {CentralPaneNavigatorParamList} from '@libs/Navigation/types';
+import * as Report from '@userActions/Report';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+
+type UserTypingEventListenerOnyxProps = {
+    /** Stores last visited path */
+    lastVisitedPath?: string;
+};
+
+type UserTypingEventListenerProps = UserTypingEventListenerOnyxProps & {
+    /** The report currently being looked at */
+    report: OnyxTypes.Report;
+};
+function UserTypingEventListener({report, lastVisitedPath}: UserTypingEventListenerProps) {
+    const didSubscribeToReportTypingEvents = useRef(false);
+    const reportID = report.reportID;
+    const isFocused = useIsFocused();
+    const route = useRoute<RouteProp<CentralPaneNavigatorParamList, typeof SCREENS.REPORT>>();
+    useEffect(() => {
+        // Ensures the optimistic report is created successfully
```

Comment:
This comment now seems pretty out of place since the new component now is focused on a "typing event listener". Can we add more details here / clarify why this line is needed? Looks like it relates to this PR: https://github.com/Expensify/App/pull/38657

Reply:
Right I do see that, I just think "the optimistic report" isn't very clear in this new component, since we don't know which "optimistic report" that it's referring to. @jjcoffee do you know what i mean here, or do you think we should just leave it and move on?

Reply:
@tienifr Do you think you can clarify/expand the comment a bit, since you're the one who originally added it too :wink:

Reply:
Haha totally fair :D

How about we update to something like:

```suggestion
        // Ensures any optimistic report that is being created (ex: a thread report) gets created and initialized successfully before subscribing
```

Reply:
@tienifr please let's make this one last change, then I believe we're good to go